providers/dns/hetzner/internal/client.go

```diff
@@ -144,17 +146,37 @@ func (c *Client) GetZoneID(domain string) (string, error) {
 }
 
 // https://dns.hetzner.com/api-docs#operation/GetZones
-func (c *Client) getZones() (*Zones, error) {
+func (c *Client) getZones(name string, pagination *Pagination) (*Zones, error) {
 	endpoint, err := c.createEndpoint("api", "v1", "zones")
 	if err != nil {
 		return nil, fmt.Errorf("failed to create endpoint: %w", err)
 	}
 
+	var values url.Values
+
+	if pagination != nil {
+		values, err = querystring.Values(pagination)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse query parameters: %w", err)
+		}
+	}
+
+	if name != "" {
+		values.Set("name", name)
+	}
```

Comment:
`url.Values` is a `map` type, and `value.Set` will crash if `pagination` is `nil`

Reply:
https://go.dev/play/p/UIUn94iY8sK

Reply:
yes I did that a bit too quickly :wink: